### ISSUE TITLE
Auto-resolve HasFactory template parameter from Laravel naming convention

### ISSUE THREAD

**Issue by alies-dev** [psalm/psalm-plugin-laravel#517]
## Description

When a model uses `use HasFactory` without specifying the generic template parameter, Psalm reports `MissingTemplateParam`. The plugin could automatically resolve the factory class using Laravel's naming convention, eliminating these false positives.

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) with plugin v4.2.0 + Psalm 7.0.0-beta17. **69 occurrences** — one per model.

```php
// app/Models/Contact.php
class Contact extends VCardResource
{
    use HasFactory; // MissingTemplateParam: expecting HasFactory<ContactFactory>
}
```

Error:
```
MissingTemplateParam: App\Models\Contact has missing template params
when extending Illuminate\Database\Eloquent\Factories\HasFactory, expecting 1
```

Every model in the project uses `use HasFactory` without the template param. This is standard Laravel — the framework resolves the factory class at runtime via naming convention.

## Laravel's Factory Resolution Convention

Laravel resolves the factory class from the model name using this logic (from `HasFactory::newFactory()`):

1. If the model has a `newFactory()` method, use its return value
2. Otherwise, derive from namespace: `App\Models\Contact` → `Database\Factories\ContactFactory`
3. The convention is: take the model's base class name, append `Factory`, place in `Database\Factories\`

## Proposed Fix

Add a handler that automatically resolves the `HasFactory` template parameter:

1. When Psalm encounters `use HasFactory` without a template param on a model class
2. Derive the factory class name using Laravel's convention: `{ModelName}Factory` in `Database\Factories\`
3. Register it as `HasFactory<Database\Factories\{ModelName}Factory>`

This would eliminate all 69 `MissingTemplateParam` errors in Monica and similar counts in virtually every Laravel app.

## Impact

Every Laravel model that uses `use HasFactory` (which is the default in all generated models) will trigger this. It's one of the most common Psalm issues in Laravel apps.
